### PR TITLE
Flash Attention 3 Support

### PR DESCRIPTION
**Pull request by warner-benjamin**
This PR adds support for the bfloat16 Flash Attention 3 path on `FlexBertUnpadRopeAttention`. FA3 only supports global attention at the moment. I haven't tried the fp8 path for FA3. Current settings are about ~2.5% faster. I expect this will increase with larger models due to more attention heads and increased hidden dim.

This PR builds on #102 and #103 and requires both of them to be merged first.

### CODE REVIEW

File: src/bert_layers/configuration_bert.py [hunk 1]
```diff
@@ -152,6 +154,8 @@ def __init__(
             local_attn_rotary_emb_dim (int | None): Rotary embedding dimension for local attention. None to disable and use `rotary_emb_dim` for all layers.
             unpad_embeddings (bool): Unpad inputs before the embedding layer.
             pad_logits (bool): Pad logits after the calculating the loss.
+            compile_model (bool): Compile the subset of the model which can be compiled.
+            masked_prediction (bool): Use only pass the masked tokens throught the final MLM layers
```
Comment (staghado) [inline src/bert_layers/configuration_bert.py:158]: nit : redundant comment : Use or pass

**Comment by staghado** [comment]
the fa3 rope part looks good(maybe some formatting to make it consistent with the rest but no big deal).
Also, some notes in the README about how to build FA3 would be appreciated.
